src/XdebugHandler.php

```diff
@@ -377,15 +377,20 @@ private function writeTmpIni(array $iniFiles, $tmpDir, &$error)
         }
 
         $content = '';
-        $regex = '/^\s*(zend_extension\s*=.*xdebug.*)$/mi';
+        $sectionRegex = '/^\s*\[(?:PATH|HOST)\s*=/mi';
+        $xdebugRegex = '/^\s*(zend_extension\s*=.*xdebug.*)$/mi';
 
         foreach ($iniFiles as $file) {
             // Check for inaccessible ini files
             if (($data = @file_get_contents($file)) === false) {
                 $error = 'Unable to read ini: '.$file;
                 return false;
             }
-            $content .= preg_replace($regex, ';$1', $data).PHP_EOL;
+            // Check and remove directives after HOST and PATH sections
+            if (preg_match($sectionRegex, $content, $matches, PREG_OFFSET_CAPTURE)) {
+                $content = substr($content, 0, $matches[0][1]);
+            }
+            $content .= preg_replace($xdebugRegex, ';$1', $data).PHP_EOL;
```

Comment:
```suggestion
            // Check and remove directives after HOST and PATH sections
            if (preg_match($sectionRegex, $data, $matches, PREG_OFFSET_CAPTURE)) {
                $data = substr($data, 0, $matches[0][1]);
            }
            $content .= preg_replace($xdebugRegex, ';$1', $data).PHP_EOL;
```
I think `preg_match/substr` should be done on `$data`, because otherwise the very last file will keep `PATH/HOST` sections and this will mess with the following code:
```php
        // Merge loaded settings into our ini content, if it is valid
        if ($config = parse_ini_string($content)) {
            $loaded = ini_get_all(null, false);
            $content .= $this->mergeLoadedConfig($loaded, $config);
        }

        // Work-around for https://bugs.php.net/bug.php?id=75932
        $content .= 'opcache.enable_cli=0'.PHP_EOL;
```

Reply:
Thanks, good catch, that is pretty sloppy! I'd better check why the test didn't flag it as well.

Reply:
https://github.com/composer/xdebug-handler/blob/c596235e84a1268e5ddb44e1b0c615846d598995/tests/Helpers/IniHelper.php#L69-L79 Loaded just in the right order :)